tensilelite/Tensile/SolutionStructs/Solution.py

```diff
@@ -2321,6 +2334,16 @@ def calSwizzleK(state, tc):
     # LDS (load size coalesced) * LSPA must load some multiple of 256 bytes.
     # No longer support loadX2/loadx4 .
     if state["DirectToLds"]:
+
+      bpeA = state["ProblemType"]["DataTypeA"].numBytes()
+      bpeB = state["ProblemType"]["DataTypeB"].numBytes()
+
+      # TODO: Currently DTL with input types of different size is not support. There are functional issues
+      # This needs to be fixed.
+      if bpeA != bpeB:
```

Comment:
I think it is better to check this after isDirectToLdsDoable and check A and B separately.
DirectToLdsA/B can be false even if DirectToLds is true.
The condition should be like
  if bpeAbefore != bpeAafter:

Reply:
> if bpeAbefore != bpeAafter:

What does before and after refer to here?

Reply:
Before conversion and after conversion.
What I thought was that like
- NN case does not use DTL for A (because TLU=True)
- then, we can use type conversion for A (like f8 to f16)
- then, bpeA != bpeB can be OK

Reply:
Good point regarding NN case, I've moved the check back to `isDirectToLdsDoable` and added test cases for mixed precision NN, TT cases.